src/main/webapp/components/api/cache-creation-controller.service.js

```diff
@@ -128,6 +128,14 @@ angular.module('managementConsole.api')
           this.createHelper(steps, address.concat('binary-keyed-jdbc-store', 'BINARY_KEYED_JDBC_STORE'), configuration['binary-keyed-jdbc-store']);
           this.createHelper(steps, address.concat('mixed-keyed-jdbc-store', 'MIXED_KEYED_JDBC_STORE'), configuration['mixed-keyed-jdbc-store']);
           this.createHelper(steps, address.concat('backup', 'BACKUP'), configuration.backup);
+
+          //if we are creating new node AUTHORIZATION we have to flag its ancestor SECURITY as new node as well
+          var auth = utils.deepValue(configuration.security, 'SECURITY.authorization.AUTHORIZATION');
+          if (utils.isNotNullOrUndefined(auth) && auth['is-new-node']) {
+            var sec = utils.deepValue(configuration.security, 'SECURITY');
+            sec['is-new-node'] = true;
+          }
+
```

Comment:
Minor nitpick, this code is identical to the code in updateConfigurationTemplate. Is it worth creating a function that can be called? Or do you expect this code to be removed when we revisit security roles?